src/Compilers/Core/Portable/PEWriter/RootModuleType.cs

```diff
@@ -194,7 +194,7 @@ bool ITypeReference.IsEnum
 
         bool ITypeReference.IsValueType
         {
-            get { throw ExceptionUtilities.Unreachable; }
```

Comment:
It was reachable before moving the filtering into the adapter layer. I'll remove now. Thanks